### PR TITLE
feature(engine): added magic widget name and description language keys

### LINKED ISSUE #3136: Remove title and description parameters from widget registration (Trac #3136)
_Original ticket http://trac.elgg.org/ticket/3136 on 41204219-09-01 by ewinslow, assigned to unknown._

Elgg version: 1.7

These are superfluous because the language strings can just be standardized: e.g.,

    widgets:$name:title
    widgets:$name:description

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #3136

### CODE REVIEW

**Comment by ewinslow** [comment]
LGTM but test failures

**Comment by jeabakker** [comment]
no more test failures

LGTM

File: docs/guides/widgets.rst [hunk 1]
```diff
@@ -50,6 +50,10 @@ Once you have created your edit and view pages, you need to initialize the plugi
 
     // Add a third file widget
     add_widget_type('filerepo3', elgg_echo("file:widget3"), elgg_echo("file:widget:description3"));
+    
+.. note::
+
+   It is not required to provide a name and description during registration of the widget if you have translations available in the following format: ``widgets:widget_handler:title`` and ``widgets:widget_handler:description``.
```
Comment (hypeJunction) [inline docs/guides/widgets.rst]: widget_handler is a variable in this case, and I think it should be clearer. Perhaps add $

File: engine/classes/Elgg/I18n/Translator.php [hunk 2]
```diff
@@ -404,10 +404,14 @@ function languageKeyExists($key, $language = 'en') {
 			return false;
 		}
 	
-		if (($language !== 'en') && !array_key_exists($language, $this->CONFIG->translations)) {
+		if (($language !== 'en') && (!isset($this->CONFIG->translations) || !array_key_exists($language, $this->CONFIG->translations))) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/I18n/Translator.php]: This doesn't belong in this commit, imho

**Comment by hypeJunction** [comment]
Not excited about more magic, but if we want this in, we should consider better namespacing. It almost feels we should not require a name and do the magic in ElggWidget::getDisplayName()...

File: engine/classes/Elgg/I18n/Translator.php (hunk 2 above)
Comment (jdalsem) [inline engine/classes/Elgg/I18n/Translator.php]: it is in a seperate commit
